plugins/pod/flytekitplugins/pod/task.py

```diff
@@ -33,7 +32,7 @@ def primary_container_name(self) -> str:
 class PodFunctionTask(PythonFunctionTask[Pod]):
     def __init__(self, task_config: Pod, task_function: Callable, **kwargs):
         super(PodFunctionTask, self).__init__(
-            task_config=task_config, task_type="sidecar", task_function=task_function, **kwargs,
+            task_config=task_config, task_type="sidecar", task_function=task_function, task_type_version=1, **kwargs,
```

Comment:
This is awesome, but as a follow up can we also create a RawPodTask? (not a PodFunctionTask), where users can just write the pod, spec and we will just run it, where they can say what is the primary container and it may look like the RawContainerTask?

Again not in this PR, maybe in the future